src/libopensc/card-openpgp.c

```diff
@@ -787,6 +788,14 @@ pgp_enumerate_blob(sc_card_t *card, struct blob *blob)
 		}
 		tag |= cla;
 
+		/* Awful hack for composite DOs that have
+		 * a TLV with the DO's id encompassing the
+		 * entire blob. Example: Yubikey Neo */
+		if (tag == blob->id) {
```

Comment:
A side effect of this hack is that the code will now treat something like the following as valid:

Reading DO 0x6E (Application Related Data):
0x6E
0xXX (some length)
...
0x73 (Discretionary data objects)
0xXX (some length)
0x73 (Discretionary data objects tag nested twice)
0xXX (some length)
0xC0 (Extended capabilities)
...

where there are two 0x73 tags nested.

Referring back to your original comment, this hack isn't YubiKey specific, because YubiKey, javacardopenpgp, and other cards all could possibly return a constructed DO with an outer layer like that.